### PR TITLE
Add license headers to all source files

### LINKED ISSUE #145: Add license information per file
Issue #9 has not been fixed, license should be added per file, giving credits to the authors, including the ones fixing issues, one of the most used legend is:

```vala
/* file-name.vala
 *
 * Copyright YEAR NAME <USER@DOMAIN>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version XXXXXX of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
```

### PR DESCRIPTION
**Pull request by Cogitri**
fixes #145

### CODE REVIEW

**Comment by benwaffle** [comment]
What is the purpose of a license header?

**Comment by Cogitri** [comment]
A license header has three main purposes:

* In case the file is obtained separately from the project (e.g. via a paste provider like pastebin) it's still clear what the license is
* It makes it easy to see who contributed to a file and has copyright
* It's actually required for clarifying whether the project is LGPL-2.1-only or LGPL-2.1-or-later . Without a license header in the source that is not clear (since the text in COPYING doesn't exactly state that). In the current version of this PR the header says: "This program is free software: you can redistribute it and/or  it under the terms of the GNU Lesser General Public License as published  the Free Software Foundation, *either version 2.1 of the License,  (at your option) any later version.*", so that'd make VLS be licensed under LGPL-2.1-or-later, meaning that if a LGPL-3.0 would ever become a thing VLS could be licensed under that as well.

**Review by Prince781 — CHANGES_REQUESTED** [review]
One change needed, but otherwise LGTM.

File: src/server.vala [hunk 1]
```diff
@@ -1,3 +1,21 @@
+/* server.vala
+ *
```
[the remark this answers is not in the document]
Comment (Cogitri) [reply src/server.vala:2]: Oh right, seems like I missed that due to the split of main.vala into this. Should be fixed now.

File: src/protocol.vala [hunk 2]
```diff
@@ -1,3 +1,23 @@
+/* protocol.vala
+ *
+ * Copyright 2017-2019 Ben Iofel <iofelben@gmail.com>
```
Comment (Cogitri) [inline src/protocol.vala:3]: Oh actually, seems like your mail address changed, @benwaffle . Do you prefer this email or `ben@iofel.me`?

Comment (benwaffle) [reply src/protocol.vala:3]: ben@iofel.me

File: test/testclient.vala [hunk 3]
```diff
@@ -1,3 +1,21 @@
+/* testclinet.vala
```
Comment (benwaffle) [inline test/testclient.vala:1]:
```suggestion
/* testclient.vala
```

**Review by benwaffle — APPROVED** [review]

**Comment by benwaffle** [comment]
why do you keep force pushing? seems easier just to push more commits

**Comment by Cogitri** [comment]
Hm, with gitlab it shows a "show diff to last version" when force pushing so I usually just force push so things end up as one commit. I guess with GitHub it's easier to make multiple commits and then do a squash merge at the end

**Comment by Prince781** [comment]
> why do you keep force pushing? seems easier just to push more commits

@benwaffle I think force-pushing is a good idea, because it keeps the history for this MR simple.

**Review by Prince781 — APPROVED** [review]

**Comment by benwaffle** [comment]
@Prince781 but you can just Squash and Merge - so there will be 1 commit on the master branch

**Comment by Prince781** [comment]
Seeing as this has two approvals, I'm merging this PR.

**Pull request merged**